lib/sentry/opentelemetry/span_processor.ex

```diff
@@ -0,0 +1,199 @@
+defmodule Sentry.OpenTelemetry.SpanProcessor do
+  @moduledoc false
+
+  require OpenTelemetry.SemConv.ClientAttributes, as: ClientAttributes
+  require OpenTelemetry.SemConv.Incubating.DBAttributes, as: DBAttributes
+  require OpenTelemetry.SemConv.Incubating.HTTPAttributes, as: HTTPAttributes
+  require OpenTelemetry.SemConv.Incubating.URLAttributes, as: URLAttributes
+  require OpenTelemetry.SemConv.Incubating.MessagingAttributes, as: MessagingAttributes
+  @behaviour :otel_span_processor
+
+  require Logger
+
+  alias Sentry.{Span, Transaction, OpenTelemetry.SpanStorage, OpenTelemetry.SpanRecord}
+
+  @impl true
+  def on_start(_ctx, otel_span, _config) do
+    span_record = SpanRecord.new(otel_span)
+
+    SpanStorage.store_span(span_record)
+
+    otel_span
+  end
+
+  @impl true
+  def on_end(otel_span, _config) do
+    span_record = SpanRecord.new(otel_span)
+
+    SpanStorage.update_span(span_record)
+
+    if span_record.parent_span_id == nil do
+      root_span_record = SpanStorage.get_root_span(span_record.span_id)
+      child_span_records = SpanStorage.get_child_spans(span_record.span_id)
+      transaction = build_transaction(root_span_record, child_span_records)
+
+      result =
+        if transaction do
+          case Sentry.send_transaction(transaction) do
+            {:ok, _id} ->
+              true
+
+            :ignored ->
+              true
+
+            {:error, error} ->
+              Logger.error("Failed to send transaction to Sentry: #{inspect(error)}")
+              {:error, :invalid_span}
+          end
+        else
+          true
+        end
+
+      :ok = SpanStorage.remove_span(span_record.span_id)
+
+      result
+    else
+      true
+    end
+  end
+
+  @impl true
+  def force_flush(_config) do
+    :ok
+  end
+
+  # TODO: "Elixir.Oban.Stager process" is an internal span and it produces a lot of noise
+  #       so it makes sense to ignore it - should this be configurable?
+  defp build_transaction(%{name: "Elixir.Oban.Stager process"}, _), do: nil
+
+  defp build_transaction(root_span_record, child_span_records) do
+    root_span = build_span(root_span_record)
+    child_spans = Enum.map(child_span_records, &build_span(&1))
+
+    Transaction.new(%{
+      span_id: root_span.span_id,
+      transaction: transaction_name(root_span_record),
+      transaction_info: %{source: :custom},
+      contexts: %{
+        trace: build_trace_context(root_span_record),
+        otel: build_otel_context(root_span_record)
+      },
+      spans: [root_span | child_spans]
+    })
+  end
+
+  defp transaction_name(
+         %{attributes: %{unquote(to_string(MessagingAttributes.messaging_system())) => :oban}} =
+           span_record
+       ) do
+    span_record.attributes["oban.job.worker"]
+  end
```

Comment:
FWIW this breaks the otel semantic conventions for messaging systems. Though the oban otel instrumentation also doesn't follow the guidelines either. This PR resolves it: https://github.com/open-telemetry/opentelemetry-erlang-contrib/pull/436

Instead of `{job} proces` it should be `process {job}` or `process {queue}`.

Reply:
@danschultzer I'm gonna hold off with Oban support until your PRs are merged, otherwise it's a moving target.